generators/python-v2/ast/src/BaseInvocation.ts

```diff
@@ -30,15 +34,32 @@ export class BaseInvocation extends AstNode {
 
     public write(writer: Writer): void {
         this.reference.write(writer);
-        writer.write("(");
 
+        if (this.arguments.length === 0) {
+            writer.write("()");
+            return;
+        }
+
+        writer.write("(");
+        if (this.multiline) {
+            writer.newLine();
+            writer.indent();
+        }
         this.arguments.forEach((arg, idx) => {
             arg.write(writer);
             if (idx < this.arguments.length - 1) {
-                writer.write(", ");
+                writer.write(",");
+                if (this.multiline) {
+                    writer.newLine();
+                } else {
+                    writer.write(" ");
+                }
             }
         });
-
+        if (this.multiline) {
+            writer.newLine();
+            writer.dedent();
+        }
```

Comment:
Is this change purely to support code snippets, or do we plan on using this function for generating actual code as well? I'm wondering if we should delegate this type of logic to formatters, instead of owning logic that deals with formatting.

Reply:
For general SDK generation, that's exactly what we do. But given that dynamic snippets need to run in the browser, we need to include some basic level of formatting for that use case.

In the long run, it's possible that we can more generally depend on WASM formatters and not lean on this type of behavior.

Reply:
> But given that dynamic snippets need to run in the browser, we need to include some basic level of formatting for that use case.

Makes sense.

> In the long run, it's possible that we can more generally depend on WASM formatters and not lean on this type of behavior.

Also makes sense.

I'm just worried about the scenario in which we end up putting a bunch of _new_ tech debt in the V2 generators.